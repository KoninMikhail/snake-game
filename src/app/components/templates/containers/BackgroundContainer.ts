import styled from 'styled-components';

/**
 * @name BackgroundContainer
 * @version 1.0.0
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Using it for place page components
 *
 * @returns {JSX.Element}
 * @constructor
 */

export const BackgroundContainer = styled.div`
    display: grid;
    position: absolute;
    align-items: center;
    justify-content: stretch;
    width: 100%;
    height: 100%;
    top: 0;
    right: 0;
    left: 0;
    overflow: hidden;
    bottom: 0;
`;
