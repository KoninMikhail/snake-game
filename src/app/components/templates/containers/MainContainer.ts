import styled from 'styled-components';
import { up } from 'styled-breakpoints';

/**
 * @name MainContainer
 * @version 1.0
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for create main content container
 *
 * @returns {JSX.Element}
 * @constructor
 */

export const MainContainer = styled.div`
    display: grid;
    position: relative;
    grid-template-rows: max-content auto max-content;
    gap: ${({ theme }) => theme.gaps.md};
    width: 100%;
    height: 100%;
    padding: 1.5em;

    ${up('md')} {
        padding: 2em;
    }

    ${up('lg')} {
        padding: 5vh 4vw;
    }
`;
