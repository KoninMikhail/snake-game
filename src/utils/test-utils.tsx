import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
