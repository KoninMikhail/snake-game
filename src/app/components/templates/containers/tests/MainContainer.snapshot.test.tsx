import { renderWithProviders } from '@utils/test-utils';
import { MainContainer } from '@templates/containers/MainContainer';

describe('MainContainer', () => {
    it('should equal snapshot', () => {
        const { container } = renderWithProviders(<MainContainer />);
        expect(container).toMatchSnapshot();
    });
});
