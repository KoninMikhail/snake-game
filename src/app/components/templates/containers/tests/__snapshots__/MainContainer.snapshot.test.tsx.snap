// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should equal snapshot 1`] = `
.c0 {
  display: grid;
  position: relative;
  grid-template-rows: max-content auto max-content;
  gap: 1rem;
  width: 100%;
  height: 100%;
  padding: 1.5em;
}

@media (min-width:768px) {
  .c0 {
    padding: 2em;
  }
}

@media (min-width:992px) {
  .c0 {
    padding: 5vh 4vw;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;
